### PR TITLE
[Core] Allow disabling ECC for nvidia-gpu

### PR DESCRIPTION
**Pull request by Michaelvll**
According to the following link, disabling ECC can improve the GPU performance by 30%, and a user has requested this knob. This is experimental and might be blocked by the task-level config, as the user would like to specify this per-task.

https://portal.nutanix.com/page/documents/kbs/details?targetId=kA00e000000LKjOCAW

Added config:
```yaml
nvidia_gpu:
  disable_ecc: true
```

We only allow disabling ecc for the clouds with SkyPilot provisioner, as those with ray autoscaler does not handle the retry correctly.

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
  - [x] `sky launch --cloud aws --gpus t4 "nvidia-smi -q | grep 'ECC Mode' -A 2"` with config set to turn off ecc.
  - [x] `sky launch --cloud gcp --gpus t4 "nvidia-smi -q | grep 'ECC Mode' -A 2"` with config set to turn off ecc.
  - [x] `sky launch --cloud kubernetes --gpus t4 "nvidia-smi -q | grep 'ECC Mode' -A 2"` with config set to turn off ecc. This will still launch the cluster though skipping the ECC disabling.
  - [x] `sky launch --cloud runpod --gpus rtxa6000 "nvidia-smi -q | grep 'ECC Mode' -A 2"` RunPod has ecc turned off by default.
  - [x] `sky launch -c test-lam --down --cloud lambda --gpus A10 "nvidia-smi -q | grep 'ECC Mode' -A 2"` skip the disabling due to the use of ray autoscaler
  - [x] `sky launch --cloud gcp --gpus t4 "nvidia-smi -q | grep 'ECC Mode' -A 2"` without config set to turn off ecc.
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

File: docs/source/reference/config.rst [hunk 1]
```diff
@@ -40,6 +40,18 @@ Available fields and semantics:
     - gcp
     - kubernetes
 
+  nvidia_gpu:
```
Comment (concretevitamin) [inline docs/source/reference/config.rst:43]: nit: nvidia_gpus?

File: docs/source/reference/config.rst [hunk 2]
```diff
@@ -40,6 +40,18 @@ Available fields and semantics:
...
+    # Disable ECC for NVIDIA GPUs (optional).
+    #
+    # Set to true to disable ECC for NVIDIA GPUs during provisioning. This is
+    # useful to improve the GPU performance in some cases (up to 30%
+    # improvement). This will only be applied if a cluster is requested with
+    # NVIDIA GPUs.
+    # https://portal.nutanix.com/page/documents/kbs/details?targetId=kA00e000000LKjOCAW.
```
Comment (concretevitamin) [inline docs/source/reference/config.rst:50]: Link seems to require login. Any other ref?

Comment (Michaelvll) [reply docs/source/reference/config.rst:50]: It is the issue with the ending `.`. Just removed it.

File: sky/templates/runpod-ray.yml.j2 [hunk 3]
```diff
@@ -52,7 +52,10 @@ setup_commands:
   # Line 'sudo grep ..': set the number of threads per process to unlimited to avoid ray job submit stucking issue when the number of running ray jobs increase.
   # Line 'mkdir -p ..': disable host key check
   # Line 'python3 -c ..': patch the buggy ray files and enable `-o allow_other` option for `goofys`
-  - sudo systemctl stop unattended-upgrades || true;
+  - {%- for initial_setup_command in initial_setup_commands %}
+    {{ initial_setup_command }}
```
[the remark this answers is not in the document]
Comment (Michaelvll) [reply sky/templates/runpod-ray.yml.j2:56]: RunPod does not support `sudo reboot`, but our commands will skip the disabling as RunPod already have ECC disabled.

File: docs/source/reference/config.rst [hunk 4]
```diff
@@ -40,6 +40,22 @@ Available fields and semantics:
     - gcp
     - kubernetes
 
+  nvidia_gpus:
+    # Disable ECC for NVIDIA GPUs (optional).
+    #
+    # Set to true to disable ECC for NVIDIA GPUs during provisioning. This is
+    # useful to improve the GPU performance in some cases (up to 30%
+    # improvement). This will only be applied if a cluster is requested with
+    # NVIDIA GPUs.
```
[the remark this answers is not in the document]
Comment (Michaelvll) [reply docs/source/reference/config.rst]: Added the comments for `best-effort`. How does that look?

**Review by romilbhardwaj — APPROVED** [review]
Thanks @Michaelvll!

**Pull request merged**